### PR TITLE
move runtime parameters into a namespace as globals 

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite
- [ ] all functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
This will address #813

**Comment by zingale** [comment]
This seems to work.  All runtime parameters are now globals, but in a namespace, `castro::`, `::gravity::`, or `diffusion::`.  To make life easier, the castro namespace is used in `Castro.H`, so we can drop the namespace qualifier for them.  But for the others, we cannot (we can't load all into the default namespace because of name collisions).

**Comment by zingale** [comment]
I think this works now on CPU and GPU.

File: Source/diffusion/diffusion_params.cpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include <AMReX_BLFort.H>
```
Comment (maxpkatz) [inline Source/diffusion/diffusion_params.cpp:1]: Perhaps we can auto-generate these files too?

**Pull request merged**